File: app/src/main/java/com/wafflestudio/snutt/adapter/SectionsPagerAdapter.java
package com.wafflestudio.snutt.adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.wafflestudio.snutt.ui.MyLectureFragment;
import com.wafflestudio.snutt.ui.NotificationFragment;
import com.wafflestudio.snutt.ui.SearchFragment;
import com.wafflestudio.snutt.ui.SettingsFragment;
import com.wafflestudio.snutt.ui.TableFragment;

/**
 * Created by makesource on 2016. 1. 16..
 */
public class SectionsPagerAdapter  extends FragmentPagerAdapter {
    /**
     * A {@link FragmentPagerAdapter} that returns a fragment corresponding to
     * one of the sections/tabs/pages.
     */
    private static final int NUM_ITEMS = 5;

    public SectionsPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int position) {
        // getItem is called to instantiate the fragment for the given page.
        switch(position) {
            case 0:
                return TableFragment.newInstance(0);
            case 1:
                return SearchFragment.newInstance(1);
            case 2:
                return MyLectureFragment.newInstance(2);
            case 3:
                return NotificationFragment.newInstance(3);
            case 4:
                return SettingsFragment.newInstance(4);
        }
        return null;
    }

    @Override
    public int getCount() {
        return NUM_ITEMS;
    }

}

File: app/src/main/java/com/wafflestudio/snutt2/adapter/SectionsPagerAdapter.kt
package com.wafflestudio.snutt2.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.wafflestudio.snutt2.ui.*

/**
 * Created by makesource on 2016. 1. 16..
 */
class SectionsPagerAdapter(fm: FragmentManager?) : FragmentPagerAdapter(fm) {
    override fun getItem(position: Int): Fragment {
        // getItem is called to instantiate the fragment for the given page.
        when (position) {
            0 -> return TableFragment.newInstance(0)
            1 -> return SearchFragment.newInstance(1)
            2 -> return MyLectureFragment.newInstance(2)
            3 -> return NotificationFragment.newInstance(3)
            4 -> return SettingsFragment.newInstance(4)
        }
        throw IllegalStateException("WTF")
    }

    override fun getCount(): Int {
        return NUM_ITEMS
    }

    companion object {
        /**
         * A [FragmentPagerAdapter] that returns a fragment corresponding to
         * one of the sections/tabs/pages.
         */
        private const val NUM_ITEMS = 5
    }
}
